src/subset.c

```diff
@@ -270,6 +270,7 @@ SEXP subsetDT(SEXP x, SEXP rows, SEXP cols) {
     ansn = LENGTH(rows);  // has been checked not to contain zeros or negatives, so this length is the length of result
     for (int i=0; i<LENGTH(cols); i++) {
       SEXP source = VECTOR_ELT(x, INTEGER(cols)[i]-1);
+      if (isNull(source)) error("Internal error: column %d of data.table is NULL; malformed", i+1);
```

Comment:
missing nocov

Reply:
It's covered though because C coverage doesn't inspect within line.